File: BackEnd/src/main/java/com/restaurant/backend/Entity/MenuItem.java
package com.restaurant.backend.Entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "menu_items")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MenuItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String name;

    private BigDecimal price;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "image_url")
    private String imageUrl;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    @Column(name = "is_available", nullable = false)
    private Boolean isAvailable = true;

    @Column(name = "preparation_time")
    private Integer preparationTime; // minutes

    private Integer calories;

    @Column(columnDefinition = "TEXT[]")
    private String[] allergens;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @OneToMany(mappedBy = "menuItem")
    @JsonIgnore
    private List<OrderItem> orderItems;
}

File: mobileapp/app/src/main/java/com/restaurant/mobileapp/data/model/MenuItem.kt
package com.restaurant.mobileapp.data.model

import com.google.gson.annotations.SerializedName
import java.math.BigDecimal

data class MenuItem(
    val id: Long? = null,
    val name: String,
    val price: BigDecimal? = null,
    val description: String? = null,
    @SerializedName("image_url")
    val imageUrl: String? = null,
    val category: Category? = null,
    @SerializedName("is_available")
    val isAvailable: Boolean = true,
    @SerializedName("preparation_time")
    val preparationTime: Int? = null,
    val calories: Int? = null,
    val allergens: Array<String>? = null
)

data class MenuItemResponse(
    val message: String,
    @SerializedName("menuItems")
    val menuItems: List<MenuItem>
)

